src/main/java/com/patchworkmc/mapping/IntermediaryHolder.java

```diff
@@ -0,0 +1,71 @@
+package com.patchworkmc.mapping;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import net.fabricmc.tinyremapper.IMappingProvider;
+
+/**
+ * A lazily loading representation of all Minecraft names and descriptors.
+ */
+public class IntermediaryHolder {
+	private final HashMap<String, Map<String, Member>> mappings = new HashMap<>();
+
+	private final IMappingProvider inverted;
+
+	private boolean loaded = false;
+
+	public IntermediaryHolder(IMappingProvider inverted) {
+		this.inverted = inverted;
+	}
+
+	public Map<String, Map<String, Member>> getMappings() {
```

Comment:
Since this is only ever used to get the mappings for a given class, we could make this `getMappings(String owner)` instead.